tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/dialogs/UIWorkingSetWizardsAuto.java

```diff
@@ -135,10 +137,18 @@ private void removeAllWorkingSets() {
 
 	private void cleanupWorkspace() {
 		try {
+			PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().closeAllEditors(false);
+			Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_BUILD, null);
```

Comment:
The PR description states `This PR Makes the Job-Manager join all jobs before finally deleting the project-root.`. However, this statement does only join the auto build job and no other jobs (like the auto refresh job, which may also conflict with resource deletion). Are you sure that the auto build job is causing the problem in this case? There is, for example `ImportTestUtils.waitForBuild()` that joins on more relevant jobs like refresh and build.

Reply:
Once again, you are correct: I believe that the reason for why this "fix" worked was that I introduced some artificial delay specifically by closing the editors (a bit like what was done in FindReplaceDialogTest for awaiting SWT-events).
I have made sure to close *all* jobs in this commit: https://github.com/eclipse-platform/eclipse.platform.ui/commit/ded182da305777b5c3fb5cfbedd88cb02449cb1f

Reply:
Note that calling `JobManager.cancel()` gives no guarantee that jobs will really be canceled. Even if they do so, the call is not blocking, so job termination may happen after the `cancel()` call returned. In particular, there will usually still be jobs running after `cancel()` has been called. See this statement in the documentation:
> Currently executing jobs will be asked to cancel but thereis no guarantee that they will do so.

So assuming your analysis of the issue is correct (and I really think it is correct and do not question it), it would be correct to join the relevant jobs (like you did before for the auto-build job). What now probably happens is that the `cancel()` call produces sufficient overhead to ensure that the conflicting job (probably auto build or auto refresh) have finished meanwhile. So at some point in time (when the overhead is not sufficient anymore), the test will fail again. I see two opportunities:
1. Re-introduce the join call, maybe also for the auto refresh job. I think this one is sufficient, still it may happen (in very, very rare cases) that the job is scheduled and started between calling `join()` and cleaning up the workspace.
2. Disable auto build for the test (see `ResourceTestUtil.setAutoBuilding()`): that's probably the cleanest solution but requires slightly more implementation effort.

Reply:
When I try to "join all threads", this is the UI that I get: 
![grafik](https://github.com/eclipse-platform/eclipse.platform.ui/assets/16443184/fa445e27-f439-4b83-baca-496ca8934e51)

Thus, I believe that indeed autobuild is the culprit. I will thus try your second solution